### DISCUSSION TITLE
How to load external program generated shell completions with zinit

### DISCUSSION THREAD

**Discussion by Cloud0310** [zdharma-continuum/zinit#648] [Q&A]
for example:
[rye](https://rye-up.com) offers a way:
```zsh
mkdir $ZSH_CUSTOM/plugins/rye
rye self completion -s zsh > $ZSH_CUSTOM/plugins/rye/_rye
```
to generate its own shell completions to use with oh-my-zsh
I wonder if zinit has similar functions builtin to achieve this gracefully

**Comment by pschmitt** [comment]
Typically you would issue the call generating the completions in the `atclone` ice. Zinit should pick it up then (provided the generated comp file is named correctly)

```zsh
zinit xxxx atclone'./rye self completion -s zsh > _rye' for xxx/rye
```
